### ISSUE TITLE
Locked Ctrl mode after Ctrl+W+W in Rust files

### ISSUE THREAD

**Issue by dev-ardi** [neovide/neovide#2228]
**Describe the bug**
After switching windows with Ctrl+W+W I get locked in a "control" mode, where every key I press is modified by control. This gets solved if I press Control again.

This doesn't seem to happen in other file types that I've checked (toml and no lsp)
This doesn't happen in plain neovim

**To Reproduce**
1. Open Rust file (with Rust Analyzer installed)
2. Open 2 windows (:vs)
3. Ctrl+W+W
4. Press k
5. Help gets shown instead of going down

**Desktop (please complete the following information):**
 ```
$ uname -a
Linux 6.6.8-arch1-1 #1 SMP PREEMPT_DYNAMIC x86_64 GNU/Linux`
$ neovide --version
neovide 0.12.0
$ nvim --version
NVIM v0.9.4
Build type: Release
LuaJIT 2.1.1702233742
```
Latest RA and latest 

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**
[neovide_rCURRENT.log](https://github.com/neovide/neovide/files/13800892/neovide_rCURRENT.log)
[neovide_r2023-12-31_12-51-53.log](https://github.com/neovide/neovide/files/13800893/neovide_r2023-12-31_12-51-53.log)
nightly compiler but it also happens in stable and beta

**Comment by fredizzimo** [comment]
This is strange, we get the ctrl key release event, but not the modifier state change event. The modifiers are reported strange in other ways as well. for example shortly after, when super is pressed it alternates between super and no modifiers state.

Are you on wayland or x11, and which desktop environment/compositor/window manager are you using?

Also since it looks like we get the wrong info from the winit library we are using, we should report a bug there once we have gathered more information. Perhaps with someone else trying to repeat it, so that we know that it’s not a system specific issue.

**Comment by dev-ardi** [comment]
I'm using X11 with i3, latest versions

**Comment by fredizzimo** [comment]
@dev-ardi, are you able to upate to latest winit 0.29.8 version in Cargo.toml?

There’s a fix that looks very relevant https://github.com/rust-windowing/winit/pull/3341, even if the bug trigger might be different.

**Comment by dev-ardi** [comment]
Do you mean building neovide with an updated version of winit?

**Comment by fredizzimo** [comment]
@dev-ardi, yes, I think it should build by just changing the version number.

**Comment by dev-ardi** [comment]
Seems to work with latest master commit :) Thanks

**Comment by dev-ardi** [comment]
Never mind, it's still not working!

**Comment by fredizzimo** [comment]
@dev-ardi, can you try disabling some plugins, and see if you find the trigger of the bug? Perhaps even `neovide -- --clean`

I reported the bug to winit as 
* https://github.com/rust-windowing/winit/issues/3388

**Comment by dev-ardi** [comment]
Unfortunately I don't have access to that environment, and I won't for a while. I'll try as soon as I can.

**Comment by fredizzimo** [comment]
This should be fixed by:
* https://github.com/neovide/neovide/pull/2443
